source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -44,23 +43,23 @@ Cluster::Cluster(
 
 void Cluster::startPreInit() {
   // If we are attaching to a pre-populated cache we need to initialize our hosts.
-  auto existing_hosts = dns_cache_->hosts();
+  // TODO: Can we just loop over the actual DNS cache map here rather than a copy?
```

Comment:
nit: use TODO(username) format

To avoid the copy could you pass in a lambda/callback to receive each host or something like that? Just throwing out some ideas.

Reply:
I implemented `iterateHostMap` in 7862a45 to address this TODO since it was easy enough.